agent/engine/docker_container_engine.go

```diff
@@ -168,7 +171,26 @@ var scratchCreateLock sync.Mutex
 
 // NewDockerGoClient creates a new DockerGoClient
 func NewDockerGoClient(clientFactory dockerclient.Factory, cfg *config.Config) (DockerClient, error) {
-	client, err := clientFactory.GetDefaultClient()
+	// Use 1.21 client if container metadata is enabled; otherwise use default
+	var err error
+	var client dockeriface.Client
+	var version dockerclient.DockerVersion
+	if cfg.ContainerMetadataEnabled {
+		version = dockerclient.Version_1_21
+		client, err = clientFactory.GetClient(version)
+		if err != nil {
+			// Retry with default
+			version = ""
+			client, err = clientFactory.GetDefaultClient()
+			seelog.Debugf("Container Metadata enabled but client version 1.21 unavailable. Using default version. Some features disabled")
```

Comment:
same comment as above and please format the message correctly. 
This has lots of useful information